torch/csrc/autograd/python_function.cpp

```diff
@@ -128,8 +119,9 @@ auto PyFunction::apply(const variable_list& inputs) -> variable_list {
     if (inputs[i].defined()) {
       input = THPVariable_Wrap(inputs[i]);
     } else {
-      THPObjectPtr tensor(_allocate_grad_output(output_info[i], _gpu_guard));
-      input = THPVariable_NewLeaf(tensor);
+      auto& info = output_info[i];
+      _gpu_guard.setDevice(info.device);
+      input = THPVariable_Wrap(info.type->zeros(info.size));
```

Comment:
How about adding `Variable VariableInfo::zeros(AutoGPU &gpu_guard)`? If you do this, then this place and one below change to `input = output_info[i].zeros(_gpu_guard)`